run_job script

hkassir72 opened the pull request:
- Updated mrun to terminate when all qflex instances terminate
- Small bug fixes to mrun
- Added run_job scripts under scripts/run_job which invokes mrun
- Moved mrun from scripts/ to scripts/run_job

Commit:
- Updated mrun to terminate when all qflex instances terminate
- Small bug fixes to mrun
- Added run_job scripts under scripts/run_job which invokes mrun
- Moved mrun from scripts/ to scripts/run_job

Hnefi reviewed (commented):
hussein, can you explain why you pushed forward the submodule versions in this commit?
e.g., there are 7 file updates in the flexus submodule, and 20 in QEMU.

Merging this PR essentially means that it brings all the changes in the submodules as well, which is not what we want unless you have added them for run_job.

Commit:
Updated flexus submodule to dev

Commit:
Updated qemu submodule to run_job

Commit:
Script Overhaul
Changed mrun to run_system, updated code structure, and added new features. Note that there is no one-to-one mapping between old and new structure.
Added validation to system and instance parsing.
Updated run_job to match run_system.
Added test_system to be used in travis tests.

Commit:
Updated bash scripts

hkassir72 commented:
In order to push this branch, qflex must support generating phases and checkpoints.

hkassir72 commented:
Run job still needs testings and deployment features to be integrated and tested before merging.
